doc/ARS/Local_ARS_HLD.md

```diff
@@ -0,0 +1,1155 @@
+
+List of scenario that will not trigger ARS re-selection
+ - Removal of interfaces from the ARS_INTERFACE table.
+ - Modification of the ARS_OBJECT associated with ARS interfaces.
+ - Once NHG is enabled through ARS, any changes to VLAN members—such as additions, deletions, or MAC address movements—will no longer trigger ARS re-selection.
+
+The diagrams below illustrate the typical sequences for ARS configuration, showcasing key workflows for both Adaptive Routing and Adaptive Switching.
+#### Sequence diagrams
+
+##### __Figure 4: Initilization flow__
+![](images/init_seq.png)
+
+1. Bind to Events
+    - Events from PortOrch
+        * The ArsOrch component binds to following events to start monitoring and processing ARS-related interfaces:
+            - Port operational status
+            - Port change
+            - Lag member change
+    - Events from RouteOrch
+        * The ArsOrch component binds to route updates to monitor nexthop group changes.
+    - Events from NhgOrch
+        * The ArsOrch component binds to nexthop group updates to monitor nexthop group changes.
+
+2. Get ARS SAI Capabilities
+    * ArsOrch retrieves ARS capabilities from the SAI layer and saves them in STATE_DB.
+
+3. Add ARS_PROFILE Table
+    * The CONFIG_DB adds the ARS_PROFILE table, which defines the ARS configuration parameters. ArsOrch uses this to create the ARS profile in SAI.
+
+4. Add ARS_OBJECT Table
+    * The CONFIG_DB adds the ARS_OBJECT table, which defines the ARS object specific configuration parameters. ArsOrch uses this to create the ARS SAI object (logically a ARS SAI object).
+
+5. ARS_NEXTHOPS Table
+    * The CONFIG_DB adds the ARS_NEXTHOPS table, which specifies which nexthops are used for identifying ARS object. If all of nexthops in added NHG, exist in ARS_NEXTHOPS and are pointing to ARS-enabled L2 interface, that NHG is eligible for ARS.
+
+6. Add ARS_INTERFACES Table
+    * The CONFIG_DB adds the ARS_INTERFACES table, causing ArsOrch to enable ARS on the corresponding ports and also used for identifying ARS object.
+  
+7. Add ARS_PORTCHANNELS Table
+    * The CONFIG_DB adds the ARS_PORTCHANNELS table, which enables ARS capabilities for the LAG and to associate LAG to alternative_path.
+
+##### __Figure 5: Interface update flow__
+![](images/if_update.png)
+
+1. Link UP Notification
+    * The PortsOrch component detects a link state change and notifies ArsOrch when a port transitions to the "UP" state.
+
+2. Set Port Scaling Factor
+    * Based on the port's speed, PortsOrch determines the appropriate scaling factor. <br>The scaling factor is then set in the SAI layer through the syncd component to adjust the port's behavior accordingly.
+
+##### __Figure 6: Nexthop group table created by RouteOrch flow__
+![](images/nhg_create_by_ro.png)
+
+1. Add ROUTE_TABLE Table
+    * The APPL_DB adds the ROUTE_TABLE table, triggering the process.
+2. Notification
+    * RouteOrch component will notify ArsOrch of NHG change.

```

Comment:
We talked about removing the notifications between orchagents rather call APIs for any handling? is this not taken care yet?

Reply:
OK